### PR TITLE
Update toolchain to nightly-2021-05-18

### PR DESCRIPTION
**Pull request by dominiccooney**
The nightly-2021-03-12 toolchain doesn't have RLS on Windows. This
breaks code completion in Visual Studio Code because the rust plugin
picks up the project toolchain.

RLS is available in all tier one platforms in nightly-2021-05-18 per:
https://rust-lang.github.io/rustup-components-history/

Signed-off-by: Dominic Cooney <dominic.cooney@gmail.com>

Update toolchain to nightly-2021-05-18

---

- [X ] `./mach build -d` does not report any errors
- [ X] `./mach test-tidy` does not report any errors (note, except tabs/spaces issue in an unrelated file... happy to fix in this PR)
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because changing the toolchain, covered by existing tests

### CODE REVIEW

**Comment by jdm** [comment]
Looks like there was a rustfmt change that needs to be applied with `./mach fmt`, too.

File: components/script_plugins/lib.rs [hunk 1]
```diff
@@ -405,7 +405,11 @@ fn match_def_path(cx: &LateContext, def_id: DefId, path: &[Symbol]) -> bool {
 fn in_derive_expn(span: Span) -> bool {
     matches!(
         span.ctxt().outer_expn_data().kind,
-        ExpnKind::Macro(MacroKind::Derive, _)
+        ExpnKind::Macro {
+            kind: MacroKind::Derive,
+            name: _,
+            proc_macro: _,
```
Comment (SimonSapin) [inline components/script_plugins/lib.rs:411]:
```suggestion
            ..
```

Nit (no need to go back for this if you’re not making other changes): unused fields in patterns can be omitted with `..`

**Comment by dominiccooney** [comment]
Ah, looks like these bots treat warnings as errors. I'll work on the errors.

**Comment by jdm** [comment]
Ah, I just fixed the arm64 issues in pathfinder yesterday. Guess I'll put out a new release.

**Comment by jdm** [comment]
@bors-servo r-
We'll need to update Cargo.lock to use pathfinder_simd 0.5.1 before this can merge.

**Comment by sagudev** [comment]
> @bors-servo retry

where is bors?

**Comment by sagudev** [comment]
Fail at `python3 -m pip install --upgrade pip virtualenv` as Release file for some ppas were not found.

**Comment by sagudev** [comment]
Maybe there should be `apt update` before, but it is also very rare occasion.

**Comment by sagudev** [comment]
@jdm do rerun

**Pull request merged**